symengine/dict.h

```diff
@@ -9,6 +9,8 @@
 #define SYMENGINE_DICT_H
 
 #include <gmpxx.h>
+#include <piranha/mp_integer.hpp>
+#include <piranha/hash_set.hpp>
```

Comment:
When I move it .cpp, I get the following error in .h
```
/home/sumith/github/csympy/symengine/dict.h:55:13: error: ‘piranha’ does not name a type
     mutable piranha::integer second;
             ^
```
which is fixed by moving the inclusion to .h